File: backend/src/modules/system/system.controller.ts
import { Controller, Get, Post, Body, Query } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { SystemService } from './system.service';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('系统管理')
@Controller('api/system')
export class SystemController {
  constructor(private readonly systemService: SystemService) {}

  // 核心信息
  @Get('info')
  @ApiOperation({ summary: '获取核心信息' })
  async getCoreInfo() {
    return ApiResponseDto.success(await this.systemService.getCoreInfo());
  }

  @Get('version')
  @ApiOperation({ summary: '获取版本' })
  async getVersion() {
    return ApiResponseDto.success(await this.systemService.getVersion());
  }

  @Get('uptime')
  @ApiOperation({ summary: '获取运行时间' })
  async getUptime() {
    return ApiResponseDto.success(await this.systemService.getUptime());
  }

  @Get('memory')
  @ApiOperation({ summary: '获取内存信息' })
  async getSharedMemory() {
    return ApiResponseDto.success(await this.systemService.getSharedMemory());
  }

  @Get('processes')
  @ApiOperation({ summary: '获取进程列表' })
  async getProcessList() {
    return ApiResponseDto.success(await this.systemService.getProcessList());
  }

  @Get('status')
  @ApiOperation({ summary: '获取系统状态' })
  async getSystemStatus() {
    return ApiResponseDto.success(await this.systemService.getSystemStatus());
  }

  @Get('modules')
  @ApiOperation({ summary: '获取模块列表' })
  async getModulesList() {
    return ApiResponseDto.success(await this.systemService.getModulesList());
  }

  // 配置管理
  @Get('config')
  @ApiOperation({ summary: '获取配置值' })
  async getConfig(@Query('group') group: string, @Query('name') name: string) {
    return ApiResponseDto.success(await this.systemService.getConfig(group, name));
  }

  @Post('config')
  @ApiOperation({ summary: '设置配置值' })
  async setConfig(@Body() body: { group: string; name: string; value: any }) {
    await this.systemService.setConfigNow(body.group, body.name, body.value);
    return ApiResponseDto.success(null, '设置成功');
  }

  @Get('config/list')
  @ApiOperation({ summary: '列出所有配置' })
  async listConfig() {
    return ApiResponseDto.success(await this.systemService.listConfig());
  }

  @Get('config/diff')
  @ApiOperation({ summary: '配置差异' })
  async diffConfig() {
    return ApiResponseDto.success(await this.systemService.diffConfig());
  }

  // TLS
  @Get('tls/list')
  @ApiOperation({ summary: '获取 TLS 连接列表' })
  async getTlsList() {
    return ApiResponseDto.success(await this.systemService.getTlsList());
  }

  @Get('tls/info')
  @ApiOperation({ summary: '获取 TLS 信息' })
  async getTlsInfo() {
    return ApiResponseDto.success(await this.systemService.getTlsInfo());
  }

  @Post('tls/reload')
  @ApiOperation({ summary: '重载 TLS' })
  async reloadTls() {
    await this.systemService.reloadTls();
    return ApiResponseDto.success(null, '重载成功');
  }

  // Pike
  @Get('pike/list')
  @ApiOperation({ summary: '获取 Pike 封禁列表' })
  async getPikeList() {
    return ApiResponseDto.success(await this.systemService.getPikeList());
  }

  @Get('pike/top')
  @ApiOperation({ summary: '获取 Pike 排行' })
  async getPikeTop(@Query('limit') limit?: number) {
    return ApiResponseDto.success(await this.systemService.getPikeTop(limit || 10));
  }

  // 统计
  @Get('statistics')
  @ApiOperation({ summary: '获取统计信息' })
  async getStatistics(@Query('group') group?: string) {
    return ApiResponseDto.success(await this.systemService.getStatistics(group));
  }

  @Post('statistics/reset')
  @ApiOperation({ summary: '重置统计' })
  async resetStatistics(@Body() body: { name: string }) {
    await this.systemService.resetStatistics(body.name);
    return ApiResponseDto.success(null, '重置成功');
  }
}

File: kam-admin-module-monitor/src/main/java/io/kamailio/admin/modules/system/controller/SystemController.java
package io.kamailio.admin.modules.system.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.modules.system.service.SystemService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Tag(name = "系统管理")
@RestController
@RequestMapping("/api/system")
public class SystemController {
    private final SystemService service;

    public SystemController(SystemService service) { this.service = service; }

    @GetMapping("/core-info") public ResponseEntity<ApiResponse<Object>> getCoreInfo() { return ResponseEntity.ok(ApiResponse.success(service.getCoreInfo())); }
    @GetMapping("/version") public ResponseEntity<ApiResponse<Object>> getVersion() { return ResponseEntity.ok(ApiResponse.success(service.getVersion())); }
    @GetMapping("/uptime") public ResponseEntity<ApiResponse<Object>> getUptime() { return ResponseEntity.ok(ApiResponse.success(service.getUptime())); }
    @GetMapping("/shmmem") public ResponseEntity<ApiResponse<Object>> getSharedMemory() { return ResponseEntity.ok(ApiResponse.success(service.getSharedMemory())); }
    @GetMapping("/psx") public ResponseEntity<ApiResponse<Object>> getProcessList() { return ResponseEntity.ok(ApiResponse.success(service.getProcessList())); }
    @GetMapping("/config") public ResponseEntity<ApiResponse<Object>> getConfig(@RequestParam String group, @RequestParam String name) { return ResponseEntity.ok(ApiResponse.success(service.getConfig(group, name))); }
    @PostMapping("/config") public ResponseEntity<ApiResponse<Void>> setConfigNow(@RequestBody MapBody body) { service.setConfigNow(body.getGroup(), body.getName(), body.getValue()); return ResponseEntity.ok(ApiResponse.success(null)); }
    @GetMapping("/config/list") public ResponseEntity<ApiResponse<Object>> listConfig() { return ResponseEntity.ok(ApiResponse.success(service.listConfig())); }
    @GetMapping("/config/diff") public ResponseEntity<ApiResponse<Object>> diffConfig() { return ResponseEntity.ok(ApiResponse.success(service.diffConfig())); }
    @GetMapping("/tls/list") public ResponseEntity<ApiResponse<Object>> getTlsList() { return ResponseEntity.ok(ApiResponse.success(service.getTlsList())); }
    @GetMapping("/tls/info") public ResponseEntity<ApiResponse<Object>> getTlsInfo() { return ResponseEntity.ok(ApiResponse.success(service.getTlsInfo())); }
    @PostMapping("/tls/reload") public ResponseEntity<ApiResponse<Void>> reloadTls() { service.reloadTls(); return ResponseEntity.ok(ApiResponse.success(null)); }
    @GetMapping("/pike/list") public ResponseEntity<ApiResponse<Object>> getPikeList() { return ResponseEntity.ok(ApiResponse.success(service.getPikeList())); }
    @GetMapping("/pike/top") public ResponseEntity<ApiResponse<Object>> getPikeTop(@RequestParam(defaultValue = "10") int limit) { return ResponseEntity.ok(ApiResponse.success(service.getPikeTop(limit))); }
    @GetMapping("/statistics") public ResponseEntity<ApiResponse<Object>> getStatistics(@RequestParam(required = false) String group) { return ResponseEntity.ok(ApiResponse.success(service.getStatistics(group))); }
    @PostMapping("/statistics/reset") public ResponseEntity<ApiResponse<Void>> resetStatistics(@RequestParam String name) { service.resetStatistics(name); return ResponseEntity.ok(ApiResponse.success(null)); }
    @PostMapping("/statistics/clear") public ResponseEntity<ApiResponse<Void>> clearStatistics(@RequestParam String name) { service.clearStatistics(name); return ResponseEntity.ok(ApiResponse.success(null)); }
    @GetMapping("/modules") public ResponseEntity<ApiResponse<Object>> getModulesList() { return ResponseEntity.ok(ApiResponse.success(service.getModulesList())); }
    @GetMapping("/status") public ResponseEntity<ApiResponse<Object>> getSystemStatus() { return ResponseEntity.ok(ApiResponse.success(service.getSystemStatus())); }

    public static class MapBody {
        private String group, name;
        private Object value;
        public String getGroup() { return group; }
        public void setGroup(String group) { this.group = group; }
        public String getName() { return name; }
        public void setName(String name) { this.name = name; }
        public Object getValue() { return value; }
        public void setValue(Object value) { this.value = value; }
    }
}
